### DISCUSSION TITLE
plugin

### DISCUSSION THREAD

**Discussion by giscus** [Yiov/vitepress-doc#3] [General]
# plugin

我的vitpress文档教程

https://vitepress.yiov.top/plugin.html

**Comment by Yiov** [comment]
[Giscus](https://vitepress.yiov.top/plugin.html#%E8%AF%84%E8%AE%BA) 搭建成功，评论测试

**Reply by mnjun9527** [reply]
牛逼啊大哥

**Comment by Aurorxa** [comment]
写的真棒

**Comment by mnjun9527** [comment]
不登录能评论吗

**Comment by soul667** [comment]
牛逼啊大哥

**Comment by cgwuli** [comment]
这文档比官方强100倍，牛逼

**Reply by cgwuli** [reply]
唯一的遗憾是，自动侧边栏还是很有用的，特别是对于有大量文档的我。强烈建议完善这个插件的样例代码，谢谢站长

**Reply by Yiov** [reply]
没有没有，官方文档还是很详细的，只是刚开始vietepress没有中文站点，不太适合上手，就尝试写了一下，现在已经rc版了，基本没什么改变了

自动侧边栏，可以看看这位作者的
https://github.com/QC2168/vite-plugin-vitepress-auto-sidebar

**Reply by cgwuli** [reply]
真心不是吹嘘！官网我最近一直看，多次看到放弃！昨天突然搜到站长你的，感觉打开一片新世界，看得人欲罢不能！你的文档最大优点：一是说人话，亲切实在。二是案例详尽，想要看的基本都有。昨天一直看到凌晨4点，跟着你的文档，轻松撸出完整的笔记，而且每一步都做得心理很有底。谢谢站长的辛苦付出！

**Reply by cgwuli** [reply]
现在唯一剩下的自动侧边栏，我看了原作者的文档，关键之处他用的注释 :)
AutoSidebar({
        // You can also set options to adjust sidebar data
        // see option document below
      })

**Reply by cgwuli** [reply]
冒昧的请求站长可否把这一部分也完善一下？我的需求是nav里有几个item，无论我选哪个item，侧边栏都是把这个nav力所有的item都渲染出来了，可否点击哪个item就渲染哪个？

**Reply by Yiov** [reply]
插件能否达到这样的效果，我也没有使用过，可以在作者那里提issue

因为现实比较忙，得闲了来看一下，我也不保证我什么时候会更一下😂

**Reply by cgwuli** [reply]
好的好的，期待......

**Reply by Yiov** [reply]
已更新另一款插件，配置简单些，可以尝试

**Reply by cgwuli** [reply]
好的，谢谢站长推荐，我去学习一下

**Reply by cgwuli** [reply]
站长好，我试了这款侧边栏插件，确实方便。但是关于默认的配置，有两个问题：1、我有多个多级nav导航栏，可这个侧边栏把所有的内容都生成了。也就是说它不能match我所选的nav的子内容。2、生成的侧边栏有标题，这很好，但其名称是我的文件夹的英文名，不能自己设置中文标题。请问这两个需求应该在哪里设置？辛苦站长

**Reply by Yiov** [reply]
1.你是希望进入某个内容页面后，侧边栏就匹配和这个内容同在一个文件夹的所有内容，是么。

那你可以看一下 [插件的高级用法](https://vitepress-sidebar.jooy2.com/advanced-usage/multiple-sidebars-how-to)

2.不想改名，又想自定义文件夹名，可以另辟蹊径解决

比如文件夹是 `assets` ，在assets下新建 `index.md` ，将 `index.md` 标题写上你要的中文标题

```
useFolderTitleFromIndexFile: true, //开启index.md显示标题
includeFolderIndexFile: false,  //关闭index.md在侧边栏显示
```

更多的功能，需要慢慢摸索

**Reply by cgwuli** [reply]
好方法，谢谢站长

**Comment by 779259529** [comment]
你好，我这边是vercel托管的，请问一下有没有对应的评论插件可用

**Reply by Yiov** [reply]
一样的呀，我代码在GitHub，用vercel导入仓库托管的

插件我是觉得giscus好用，数据都在仓库里，你觉得不好用，可以尝试我列举的其他插件

**Comment by liueic** [comment]
busuanzi好像用不了了

**Reply by Yiov** [reply]
我也觉得不咋好用，找找其他的吧

**Reply by Yiov** [reply]
找了一圈，发现还是用它吧，现在我封装了一下，偶尔抽风不显示，也正常的，免费的要啥自行车

**Comment by lljxww** [comment]
学到了很多, 谢谢站长!

**Reply by Yiov** [reply]
客气了，就是互相学习，简单入个门啦

**Comment by yangBron** [comment]
您好，图片缩放覆盖不了侧边栏和导航栏，

**Reply by Yiov** [reply]
那你可以将值调大一点

```diff
/* .vitepress/theme/style/var.css */
.medium-zoom-overlay {
-    z-index: 20;
+   z-index: 30;
 }
```

**Reply by yangBron** [reply]
我的情况是 图片足够大，但是被侧边栏和导航栏覆盖了，

**Reply by Yiov** [reply]
太大了，用margin缩小就行了，数值越大，缩放的越小；background留空，就没有背景

```diff
/* .vitepress/theme/index.ts */
- mediumZoom('.main img', { background: 'var(--vp-c-bg)' });
+ mediumZoom('.main img', { margin:100, background: '' }); 
```

**Comment by cgwuli** [comment]
站长好，我看您的网站设置了多语言，发现了一个bug。如果是在首页，点击切换语言没有问题。但是如果是在内容页，点击切换语言会出现404，我想这应该是没有做对应的英文版内容所导致的吧。这里我有个想法，能否改进多语言的功能---无论处在哪个页面，当我们点击多语言时，都导航到对应语言的首页，这样就能防止出现404了。说一下为什么不用下拉菜单实现呢？因为还需要多语言的只搜索对应语言的内容这个功能。

**Reply by cgwuli** [reply]
站长您看看能否实现？

**Reply by Yiov** [reply]
那不是bug，正如你所想，我就是没有放对应的文件，所以导致404

我只是做个简单的入门，翻译后再放入工作量太大，简单演示一下就行

你只要自己去实践过知道，语言切换文章也是对应切换的，[官网国际化](https://vitepress.dev/zh/guide/i18n) 都有说明

**Reply by cgwuli** [reply]
明白了，谢谢站长

**Comment by HM-Suiji** [comment]
Giscus评论必须选择自己文档的仓库作为公开的吗，文档仓库只能私有咋办

**Reply by Yiov** [reply]
可以选择其他仓库，单独新建一个专门放评论，但是必须要公开

非公开的，只能选择其他评论系统，自建后端

**Reply by nxcare** [reply]
我也是文档仓库是私有的，按作者的办单独新建了一个仓库，配置好以后，在文档后面显示，错误：giscus is not installed on this repository
是不是只能在本文档仓库生效啊

**Reply by Yiov** [reply]
评论放哪个仓库，就在安装giscus后，选择放评论的仓库即可

**Reply by nxcare** [reply]
我是新建了一个公开仓库，然后安装giscus后，选择了放评论的仓库，但是提示giscus is not installed on this repository

**Reply by nxcare** [reply]
已弄好，感谢作者

**Comment by dakeAa123456** [comment]
大神，@xinlei3166/vitepress-theme-website 的 Live2D 插件  使用这个插件的waline，在github部署打包报错，能帮帮忙抽空看下吗？
错误原因：Unknown file extension ".css" for /home/runner/work/waline/waline/node_modules/.pnpm/@waline+client@3.3.2/node_modules/@waline/client/dist/waline.css
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for /home/runner/work/waline/waline/node_modules/.pnpm/@waline+client@3.3.2/node_modules/@waline/client/dist/waline.css

**Reply by Yiov** [reply]
因为waline更新了，而作者插件还没有更新，所以要么等作者修，要么参照官网自己搭建，waline官网有教程，其实也不难的

**Comment by aipeach** [comment]


**Reply by Yiov** [reply]
写在Vue组件中引入就可以了，具体的你看一下Vuejs的示例就明白了

这是 [Vuejs的主题页面](https://cn.vuejs.org/ecosystem/themes.html)，它完全是从json中获取的

这是官网的[仓库代码](https://github.com/vuejs/docs/blob/main/src/ecosystem/themes/ThemeList.vue)

```diff
<script setup lang="ts">
+ import themes from './themes.json'
import ThemeListItem from './ThemeListItem.vue'
</script>

<template>
  <div class="ThemeList">
    <div class="container">
+      <div v-for="provider in themes" :key="provider.name" class="item">
        <ThemeListItem :provider="provider" />
      </div>
    </div>
  </div>
</template>
```

**Comment by bigmanBass666** [comment]
好帅啊那张龙图, 专门去搜了Xision😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍😍

**Comment by zxyang3636** [comment]
请问大佬，我的vitepress是部署在服务器的nginx上，在本地运行时评论是正常的可以登录，但是一放到服务器上就评论登录不了，显示404 Not Found nginx/1.27.3 请问咋解决

**Reply by Yiov** [reply]
要么是国内服务器网络限制了，要么是nginx的配置问题吧

**Reply by Aurorxa** [reply]
没备案吧，也有可能配置有问题

**Comment by Seasir-Hyde** [comment]
6666

**Comment by Keli365** [comment]
测试

**Comment by cgwuli** [comment]
大佬好，看到您更新了自动侧边栏插件VitePress Sidebar，我又试了一下。看了文档后还是有两个问题：
1、插件提供了两个函数：withSidebar和generateSidebar，官方推荐用withSidebar，我试了一下，可以对于docs目录中的所有文件自动生成侧边栏，但是好像不能与导航栏联动生成对应的目录的侧边栏。
2、如果想与导航栏联动，是否只能使用generateSidebar？我也试了一下，可以与导航栏联动生成侧边栏，但是在导航栏中跳转会404
大佬有时间的话，是否能提供一下用例？感激！

**Reply by cgwuli** [reply]
更正：
2、如果想与导航栏联动，是否只能使用generateSidebar？我也试了一下，可以与导航栏联动生成侧边栏，但是在 “侧边栏” 中跳转会404

**Reply by Yiov** [reply]
本人仅编写文档，这些问题我无法解答。自动侧边栏插件是使用的 [@jooy2/vitepress-sidebar](https://github.com/jooy2/vitepress-sidebar) 插件，你可以到作者大佬的仓库提issues进行交流

**Reply by cgwuli** [reply]
好的，从您的文档中受益很多，谢谢大佬

**Comment by caolib** [comment]
todo插件配置那写错了吧，你导入taskLists然后下面使用markdownItTaskCheckbox，肯定报错啊

```ts
// .vitepress/config.mts
import markdownItTaskCheckbox from "markdown-it-task-checkbox";

export default defineConfig({
  markdown: {
    config: (md) => {
      md.use(markdownItTaskCheckbox) //todo
    }
  },
})
```

**Reply by Yiov** [reply]
感谢指正，已订正文章错误。

**Comment by cgwuli** [comment]
大佬我有一个问题，跟着您的指南一步步做到尾声了，接下来想买个轻量服务器学习一下网站部署。咨询了阿里和腾讯，目前做网站貌似政策管的很严，需要实名还有限制，比如说不能搞论坛留言之类。您这个giscus留言的功能合规吗？如果我也做的话（编程语言的学习网站），会不会封甚至拉入黑名单？

**Reply by Yiov** [reply]
申请的时候只要说是个人日记，别说什么博客论坛就行，一般很容易通过。giscus影不影响不清楚，反正文章尽量不要有政治问题，翻墙之类的，一屏蔽一个不吱声

**Reply by cgwuli** [reply]
哈哈哈，明白了，谢谢大佬指点

**Comment by nxcare** [comment]
切换路由进度条有个bug，比如你的网盘，切换时在右上角的微信处有个遮罩一闪而过
